### PR TITLE
MYLUTUSERS-1 : Initial commit

### PR DESCRIPTION
**Pull request by vanhemsj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,17 @@
-# lutece-auth-module-mylutece-users
+![](https://dev.lutece.paris.fr/jenkins/buildStatus/icon?job=module-myusers-deploy)
+[![Alerte](https://dev.lutece.paris.fr/sonar/api/project_badges/measure?project=fr.paris.lutece.plugins%3Amodule-mylutece-users&metric=alert_status)](https://dev.lutece.paris.fr/sonar/dashboard?id=fr.paris.lutece.plugins%3Amodule-mylutece-users)
+[![Line of code](https://dev.lutece.paris.fr/sonar/api/project_badges/measure?project=fr.paris.lutece.plugins%3Amodule-mylutece-users&metric=ncloc)](https://dev.lutece.paris.fr/sonar/dashboard?id=fr.paris.lutece.plugins%3Amodule-mylutece-users)
+[![Coverage](https://dev.lutece.paris.fr/sonar/api/project_badges/measure?project=fr.paris.lutece.plugins%3Amodule-mylutece-users&metric=coverage)](https://dev.lutece.paris.fr/sonar/dashboard?id=fr.paris.lutece.plugins%3Amodule-mylutece-users)
+
+# Module MyLutece users
+
+## Introduction
```
Comment (seboo) [inline README.md:8]: Please fill all the Xdoc sections

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            <releases>
+                <enabled>false</enabled>
+            </releases>
+        </repository>
+    </repositories>
+    <dependencies>
+        <dependency>
+            <groupId>fr.paris.lutece</groupId>
+            <artifactId>lutece-core</artifactId>
+            <version>[7.0.0,)</version>
```
Comment (seboo) [inline pom.xml:36]: is it really compatible only with lutece 7 ?

File: src/java/fr/paris/lutece/plugins/mylutece/modules/users/resources/users_messages.properties [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Plugin's keys
+module.provider=Mairie de paris
+module.description=mylutece users module
+# Admin features keys
+adminFeature.myLuteceUsersManagement.name=Front-offce users management
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/mylutece/modules/users/resources/users_messages.properties]: (a letter is missing)

File: src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+     * @param nIdLocalUser
+     *            The localUser id
+     */
+    public static void doLocalUserAssignements( List<String> myLuteceRoleKeysList, int nIdLocalUser )
+    {
+
+        List<LocalUserRole> localUserRolesList = LocalUserRoleHome.getLocalUserRolesListByUserId( nIdLocalUser );
+        if ( myLuteceRoleKeysList == null )
+        {
+            LocalUserRoleHome.removeFromLocalUserRoleList( localUserRolesList );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java]: perhaps  this method could be named "removeLocalUserRoles" ?

File: src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                .filter( myLuteceRoleKey -> !localUserRolesList.stream( ).anyMatch( localUserRole -> myLuteceRoleKey.equals( localUserRole.getRoleKey( ) ) ) )
+                .collect( Collectors.toList( ) );
+
+        if ( localUsersRolesToDeleteList != null )
+        {
+            LocalUserRoleHome.removeFromLocalUserRoleList( localUsersRolesToDeleteList );
+        }
+        if ( myLuteceRoleKeyToAssignList != null )
+        {
+            LocalUserRoleHome.createFromLocalUserRoleKeyList( myLuteceRoleKeyToAssignList, nIdLocalUser );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java]: this method cuold be named "createLocalUserRoles"

File: src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
+--
+-- Structure for table mylutece_users_localuser
+--
+DROP TABLE IF EXISTS mylutece_users_localuser;
+CREATE TABLE mylutece_users_localuser (
+id_local_user int AUTO_INCREMENT,
+login varchar(255) default '' NOT NULL,
+given_name varchar(255) default '' NOT NULL,
+family_name varchar(255) default '' NOT NULL,
```
Comment (seboo) [inline src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql]: "last_name" could be more accurate

File: src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql [hunk 7]
```diff
@@ -0,0 +1,22 @@
+--
+-- Structure for table mylutece_users_localuser
+--
+DROP TABLE IF EXISTS mylutece_users_localuser;
+CREATE TABLE mylutece_users_localuser (
+id_local_user int AUTO_INCREMENT,
+login varchar(255) default '' NOT NULL,
```
Comment (seboo) [inline src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql:7]: "connect_id" or "user_name" should be more coherent with Lutece history

File: webapp/WEB-INF/templates/admin/plugins/mylutece/modules/users/manage_localuserroles.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        <@messages infos=infos />
+        <div class="clearfix"></div>
+        <@tform name='action_doassignlocaluserrole'
+            action='jsp/admin/plugins/mylutece/modules/users/ManageLocalUserRoles.jsp'>
+            <input type="hidden" id="id_localuser" name="id_localuser" value="${id_localuser}" />
+            <@row>
+                <@columns>
+                    <@listGroup>
+                        <#list mylutece_role_list as role>
+                            <#if role.role !="none">
```
Comment (seboo) [inline webapp/WEB-INF/templates/admin/plugins/mylutece/modules/users/manage_localuserroles.html:19]: it is not a good practice to create a "none" role to represent an empty list

**Review by seboo — APPROVED** [review]
Please apply the small corrections requested

File: README.md (hunk 1 above)
Comment (vanhemsj) [reply README.md:8]: done

File: pom.xml (hunk 2 above)
Comment (vanhemsj) [reply pom.xml:36]: yes

File: src/java/fr/paris/lutece/plugins/mylutece/modules/users/resources/users_messages.properties (hunk 3 above)
Comment (vanhemsj) [reply src/java/fr/paris/lutece/plugins/mylutece/modules/users/resources/users_messages.properties]: done

File: src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java (hunk 4 above)
Comment (vanhemsj) [reply src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java]: done

File: src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java (hunk 5 above)
Comment (vanhemsj) [reply src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java]: done

File: src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql (hunk 6 above)
Comment (vanhemsj) [reply src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql]: ok :/

File: src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql (hunk 7 above)
Comment (vanhemsj) [reply src/sql/plugins/mylutece/modules/users/plugin/create_db_mylutece-users.sql:7]: ok :/

File: webapp/WEB-INF/templates/admin/plugins/mylutece/modules/users/manage_localuserroles.html (hunk 8 above)
Comment (vanhemsj) [reply webapp/WEB-INF/templates/admin/plugins/mylutece/modules/users/manage_localuserroles.html:19]: impossible.

**Pull request merged**